protocol/communities/manager.go

```diff
@@ -4701,6 +4701,10 @@ func (m *Manager) AddCommunityToken(token *community_token.CommunityToken) (*Com
 		if err != nil {
 			return nil, err
 		}
+
+		if token.PrivilegesLevel == community_token.OwnerLevel {
+			m.promoteSelfToControlNode(community)
+		}
```

Comment:
@mprakhov I am analyzing the code in that area now, and after deeper thought, I'm struggling to understand why this change was necessary. Shouldn't only the control node be able to mint the owner token? This means the node has to already be a control node at that stage and `m.promoteSelfToControlNode(community)` is redundant :thinking:

Reply:
Due to the signature of the CommunityDescription. When we create the community, the community private key is different from the control node.
So when we deploy the contract owner (receiving owner token) - we change the community private key to be a private key of the contract owner. Otherwise, when you extract the signer through `unwrapCommunityDescriptionMessage` it will be signed by the community private key and you'll not pass `HandleCommunityDescriptionMessage` due to `ErrNotAuthorized` error

Reply:
So yes, only the control node will mint the owner token, but as soon as we mint the owner token, all CommunityDescriptions verifications will be done through the loop, and loop check chat contact owner and signer of the CommunityDescription was always the same

Reply:
Right, you're correct. Sorry, I had a brain fart :sweat_smile: